frontend/src/scenes/dashboard/dashboardLogic.tsx

```diff
@@ -885,7 +861,9 @@ export const dashboardLogic = kea<dashboardLogicType>([
             let refreshesFinished = 0
             let totalResponseBytes = 0
 
-            const hardRefreshWithoutCache = ['refresh', 'load_missing'].includes(action)
+            const hardRefreshWithoutCache = ['refresh', 'load_missing', 'refresh_insights_on_filters_updated'].includes(
```

Comment:
That was because upon changing filters, a request to /insight is made with refresh=false which might return `null` as results. Maybe that's different with your other changes now? But yeah let's check tomorrow.